caching-service/src/main/java/org/zowe/apiml/caching/service/vsam/VsamFile.java

```diff
@@ -17,31 +17,59 @@
 import org.zowe.apiml.zfile.*;
 
 import java.io.Closeable;
+import java.io.UnsupportedEncodingException;
+import java.util.*;
 import java.util.regex.Pattern;
 
 /**
  * ZFile wrapper providing convenience methods and implementing Closeable interface
- *
- * Creates a proxy of com.ibm.jzos.ZFileException and wraps it's methods
- *
+ * This class is intended for serialized access to VSAM file.
+ * Concurrency is to be handled by retrying.
+ * Creates a proxy of com.ibm.jzos.ZFileException and provides high level methods for CRUD operations
  */
 
 @Slf4j
-public class VsamFile implements Closeable, ZFile {
+public class VsamFile implements Closeable {
 
     private ZFile zfile;
     private VsamConfig vsamConfig;
-    private String options = "ab+,type=record";
+    private final VsamConfig.VsamOptions options;
+
+    public static final String VSAM_RECORD_ERROR_MESSAGE = "VsamRecordException occured: {}";
+    public static final String RECORD_FOUND_MESSAGE = "Record found: {}";
+    public static final String UNSUPPORTED_ENCODING_MESSAGE = "Unsupported encoding: {}";
+
     private static final Pattern REGEX_CORRECT_FILENAME = Pattern.compile("^\\/\\/\\'.*'");
 
-    public VsamFile(VsamConfig config) {
+    public VsamFile(VsamConfig config, VsamConfig.VsamOptions options) {
+        this(config, options, false);
+    }
+
+    public VsamFile(VsamConfig config, VsamConfig.VsamOptions options, boolean performWarmup) {
         if (config == null) {
             throw new IllegalArgumentException("Cannot create VsamFile with null configuration");
         }
+
         this.vsamConfig = config;
+        this.options = options;
+        log.info("VsamFile::new with parameters: {}, Vsam options: {}", this.vsamConfig, this.options);
+
+        if (!REGEX_CORRECT_FILENAME.matcher(vsamConfig.getFileName()).find()) {
+            throw new IllegalArgumentException("VsamFile name does not conform to //'VSAM.DATASET.NAME' pattern");
+        }
+
         try {
             this.zfile = openZfile();
-        } catch (ZFileException e) {
+
+            if (performWarmup) {
+                log.info("Warming up VSAM file");
+                warmUpVsamFile();
+            } else {
+
```

Comment:
Maybe add a comment if it is intentionally empty?

Reply:
good catch, I removed it